dotCMS/src/main/java/com/dotcms/contenttype/business/FieldFactoryImpl.java

```diff
@@ -88,6 +89,19 @@ public Field byContentTypeIdFieldVar(String id, String var) throws DotDataExcept
   public List<Field> byContentTypeVar(String var) throws DotDataException {
     return selectByContentTypeVarInDb(var);
   }
+    @Override
+    public List<FieldVariable> byFieldVariableKey(final String key) throws DotDataException {
+        DotConnect dc = new DotConnect();
+        dc.setSQL(sql.selectFieldVarByKey);
+        dc.addParam(key);
+
+        List<Map<String, Object>> results = dc.loadObjectResults();
+        if (results.size() == 0) {
```

Comment:
this could be wrote such as:
```
return UtilMethods.isSet(results)?new DbFieldVariableTransformer(dc.loadObjectResults()).asList(): Collections.emptyList();
```

Reply:
Done